{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.7.6","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"markdown","source":"# Tutorial: Preprocessing Deep Learning Input from RNA Strings\n\n# 1. Why bother with RNA properties prediction?\n\nRNA Technology is now a new promising hope in Health Science. In addtion to the global-scaled use of COVID19 mRNA-vaccines from BioNTech/Pfizer and Moderna, RNA drugs also provides [a new hint on how to cure important disease such as cancers](https://jhoonline.biomedcentral.com/articles/10.1186/s13045-020-00951-w).\n\nSerious learners about RNA should take this [amazing course](https://www.edx.org/course/introduction-to-biology-the-secret-of-life-3), or just look at this [cute video](https://youtu.be/JQByjprj_mA) (and the others in this channel) if you don't have much time. \n\nLeading labs including [BioNTech](https://biontech.de/how-we-translate/mrna-therapeutics), [Moderna](https://www.modernatx.com/mrna-technology/mrna-platform-enabling-drug-discovery-development) and [others](https://www.accenttx.com/our-scientific-focus/rmps-and-cancer/) state that RNA-related technology has an important advantage over DNA-technology such that it will not provide a **permanent effects** on a patient, i.e. if we modify our genes in DNA, the modification will permanently embed in the DNA, including any modification mistakes.\n\nIn contrast, injected RNA will do its job i.e. making protiens, and then completely disappeared, without leaving any permanent effects. Proteins made by RNA could prevent some\nimportant diseases such as in the case of mRNA vaccines. Intuitively, Moderna compares DNA as a life \"hardware\" while RNA as a \"software\" which can be safely injected to human body.\n\nTo design RNA drug or vaccine, some properties of RNA must be known. In the past, these properties have to be tested in bio labs which can be time and resource consuming.\nTherefore, it is difficult to conduct experiments in a large scale e.g. millions of RNAs. \n\nDeep learning technology then can help predicting specific properties of millions of RNAs in a flash. Even though, its predictions are not entirely 100% accurate, it can serve as a first filter\nfor biologists to select most promising samples to further make a small-scale intensive case in the next phase.\n\n\n## 1.1 About this Tutorial\nThis interactive tutorial notebook demonstrates how to preprocess RNA data into an input ready to feed into Deep Learning models.\nThe processed data will be similar to the ones provided by [Kaggle's mRNA OpenVaccine competition](https://www.kaggle.com/c/stanford-covid-vaccine/overview).\nSo that once you finish processing your own RNA data, you can use any of the [great public codes](https://www.kaggle.com/c/stanford-covid-vaccine/code?competitionId=22111&sortBy=voteCount) \nto make a prediction. We will provide a simplified SOTA top-solution, which will be easily reproducible and adapt to your own work in the near future, so stay tune!!\n\nFigure 1 shows the overall big picture of this notebook. \n\nHere, we assume that the user has a list of \"RNA strings\" as input where an RNA string is a string consisting of only 4 characters 'ACGU'.\nThe true RNA structure of a given string is unknown as can be illustrated in Figure 1 (bottom-left).\nThe user wants to extract this unknown structure and represents it in the format which is ready-to-use on a Deep Learning model, e.g. a neural network.\n\n\nIn this tutorial, this unknown structure will be estimated using various python and bioinformatics libraries. \nTo feed into a deep learning model, the estimated structure will be best represented by \"Graph data\" $G = (X, E)$ where\neach node in the graph corresponding to each character in a string.\n\nNote that this **Graph Representation** generalizes **Sequential Data Representation** so while you can still use well-known\nsequential layers/models such as '1D Convolution', 'Transformers' or 'Recurrent Neural Networks', you can also use a new frontier\nlike 'Graph Neural Networks' which is actually shown effective in this RNA data.\n\n$X$ is a sequence of **node feature vectors**,\nand $E$ is an adjacency matrix specifying linkages for each pair of nodes whether the pair is connected in the true structure.\n\nIn general, each element in an adjacency matrix can be a real number (e.g. specifying a \"probability\" whether a pair is connected).\nMoreover, there can be more than one adjacency matrices (e.g. differently estimated from various libraries). \n\nIn this general setting of having a list of several adjacency matrices, we will shortly call $E$ as a list of **edge features**.\n\n**Acknowledgement** this tutorial is built on amazing works of many kagglers especially [@its7171](https://www.kaggle.com/its7171/how-to-generate-augmentation-data) and [@group16](https://www.kaggle.com/group16/generating-bpps-with-arnie). However, the URLs and packages used in their notebooks are already outdated. This tutorial fix the problems, ensure complete reproducibility using static Kaggle's dataset, as well as providing full detailed explanation with minimum requirements on the input data.\n","metadata":{"papermill":{"duration":0.037582,"end_time":"2022-01-30T13:02:14.657954","exception":false,"start_time":"2022-01-30T13:02:14.620372","status":"completed"},"tags":[]}},{"cell_type":"markdown","source":"<!-- ![Data preprocessing from an RNA string](https://i.ibb.co/8mkQ1vh/RNA-data-preprocessing.png) -->\n<img src=https://i.ibb.co/8mkQ1vh/RNA-data-preprocessing.png width=\"1024\">\nFigure 1. Data preprocessing from an RNA string","metadata":{"papermill":{"duration":0.034986,"end_time":"2022-01-30T13:02:14.732168","exception":false,"start_time":"2022-01-30T13:02:14.697182","status":"completed"},"tags":[]}},{"cell_type":"markdown","source":"\n## 1.1 Overview of this tutorial\n\nWe will assume that users have already list of RNA strings to extract the graph features explained above. Here, we will use Kaggle's OpenVaccine dataset containing 6,000 RNA strings. Note that in OpenVaccine, the dataset already contained some of node and edge features. We will not use them in this tutorial but will show how to extract features even beyond ones in OpenVaccine.\n\n**Node Features.** Each character in an RNA string is represented by its character \"A, C, G or U\" and its loop-type as shown in Figure 1 \"Exterior, Bulge, Stem, Multibranch or Hairpin\". We will represent each character by 1-hot vector $R^4$ and will represent the loop-type vector in $R^6$ since there are 6 possible loop types. \n\nNote that RNA structure can be very complex and have a wide variety as shown in Figure 2.\n\n![](https://i.ibb.co/swD1DBF/rna-folding-examples.png)\nFigure 2. RNA can have a variety of structure due to folding.\n\nTherefore, due to complexity of the RNA folding structure, exact loop-type property is unknown (already mentioned in Figure 1), the `CapR` library we used can estimate loop-type probability. Therefore, it will return a probability vector in $R^6$.\n\nTherefore, by concatenating the two vectors, our node features will lies in $R^{10}$. Note that it's certainly possible that to also employ other libraries to extract more node features so that our node-feature vector lies in a higher dimensional than $R^{10}$.\n\n\n**Edge Features.** Similar to loop-type features, exact link between a pair of nodes is unknown. Therefore, **\"base-pair probabilities matrix (bpps)\"** is extracted instead. Suppose that an RNA string is length of $N$, bpps then is in $R^{N \\times N}$.\n\nThis tutorial show how to extract edge features using bpps libraries such as `vienna`, `contrafold` and `rnastructure`. From bpps of these libraries, we can also get \"most probable\" structure, which can also be further extracted as either node or edge feature.\n\nAfter the node and edge features are extracted, they can be input directly to deep learning models as shown in Figure 3. The RNA Deep Learning notebook tutorial will be published soon.\n\n<img src=https://i.ibb.co/r3WB24R/RNA-feature-and-model.png width=\"750\">\nFigure 3. Extracted feature can be used directly to models such as LSTM, GRU, Transformers and Graph Neural Networks (GNNs)\n","metadata":{"papermill":{"duration":0.032183,"end_time":"2022-01-30T13:02:14.802027","exception":false,"start_time":"2022-01-30T13:02:14.769844","status":"completed"},"tags":[]}},{"cell_type":"markdown","source":"## 1.2 Install necessary external libraries\n\nAs mention above, we will install `CapR`, `vienna`, `contrafold` and `rnastructure` packages to extract node and edge features.\nFor convenient, We will also install the package called `arnie` which provides a unified interface on all bpps libraries mentioned above.","metadata":{"papermill":{"duration":0.032612,"end_time":"2022-01-30T13:02:14.86935","exception":false,"start_time":"2022-01-30T13:02:14.836738","status":"completed"},"tags":[]}},{"cell_type":"markdown","source":"\n### Important Note on Kaggle's Docker Version\nThis kaggle notebook's environment is frozen, not to update to the latest docker version.\nThe libraries used in this notebook are also stored as Kaggle's dataset and work perfectly in the current frozen environment.","metadata":{"papermill":{"duration":0.0335,"end_time":"2022-01-30T13:02:14.935334","exception":false,"start_time":"2022-01-30T13:02:14.901834","status":"completed"},"tags":[]}},{"cell_type":"code","source":"WORKING_DIR = '/kaggle/working/'\nTOOLKITS_DIR = '/kaggle/input/rna-data-preprocessing-toolkits'\n!ls {TOOLKITS_DIR}\n!/opt/conda/bin/python3.7 -m pip install --upgrade pip","metadata":{"_kg_hide-output":true,"papermill":{"duration":12.395813,"end_time":"2022-01-30T13:02:27.366586","exception":false,"start_time":"2022-01-30T13:02:14.970773","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:19:06.379416Z","iopub.execute_input":"2022-02-13T09:19:06.379865Z","iopub.status.idle":"2022-02-13T09:19:18.872423Z","shell.execute_reply.started":"2022-02-13T09:19:06.379824Z","shell.execute_reply":"2022-02-13T09:19:18.871528Z"},"collapsed":true,"jupyter":{"outputs_hidden":true},"trusted":true},"execution_count":1,"outputs":[]},{"cell_type":"markdown","source":"The benefit of using kaggle notebook over alternative notebook such as Colab is the availability of datasets for everybody.\nHere, all external libraries are stored in a Kaggle dataset on a path specified by`TOOLKITS_DIR`. Firstly, we install `vienna` package (ref: https://www.tbi.univie.ac.at/RNA/)\n","metadata":{"papermill":{"duration":0.03717,"end_time":"2022-01-30T13:02:27.442176","exception":false,"start_time":"2022-01-30T13:02:27.405006","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!apt-get install {TOOLKITS_DIR}/viennarna_2.4.15-1_amd64.deb -y","metadata":{"_kg_hide-input":true,"_kg_hide-output":true,"papermill":{"duration":6.914188,"end_time":"2022-01-30T13:02:34.393656","exception":false,"start_time":"2022-01-30T13:02:27.479468","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:19:18.875201Z","iopub.execute_input":"2022-02-13T09:19:18.875523Z","iopub.status.idle":"2022-02-13T09:19:25.689385Z","shell.execute_reply.started":"2022-02-13T09:19:18.875489Z","shell.execute_reply":"2022-02-13T09:19:25.688424Z"},"collapsed":true,"jupyter":{"outputs_hidden":true,"source_hidden":true},"trusted":true},"execution_count":2,"outputs":[]},{"cell_type":"markdown","source":"The `CapR` package is already compiled and executable. It is retrieved from https://github.com/fukunagatsu/CapR","metadata":{"papermill":{"duration":0.048367,"end_time":"2022-01-30T13:02:34.491271","exception":false,"start_time":"2022-01-30T13:02:34.442904","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!cp -rf {TOOLKITS_DIR}/CapR/CapR/ .\n!chmod 755 ./CapR/CapR","metadata":{"_kg_hide-input":true,"_kg_hide-output":true,"papermill":{"duration":2.231437,"end_time":"2022-01-30T13:02:36.771353","exception":false,"start_time":"2022-01-30T13:02:34.539916","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:19:25.690872Z","iopub.execute_input":"2022-02-13T09:19:25.691319Z","iopub.status.idle":"2022-02-13T09:19:28.048946Z","shell.execute_reply.started":"2022-02-13T09:19:25.691271Z","shell.execute_reply":"2022-02-13T09:19:28.047464Z"},"jupyter":{"source_hidden":true},"trusted":true},"execution_count":3,"outputs":[]},{"cell_type":"markdown","source":"Next, the `arnie` package:   https://github.com/DasLab/arnie","metadata":{"papermill":{"duration":0.048911,"end_time":"2022-01-30T13:02:36.869058","exception":false,"start_time":"2022-01-30T13:02:36.820147","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!cp -rf {TOOLKITS_DIR}/arnie/arnie .\n!cp -rf {TOOLKITS_DIR}/draw_rna_pkg/draw_rna_pkg .\n!cd draw_rna_pkg && python setup.py install","metadata":{"_cell_guid":"79c7e3d0-c299-4dcb-8224-4455121ee9b0","_kg_hide-input":true,"_kg_hide-output":true,"_uuid":"d629ff2d2480ee46fbb7e2d37f6b5fab8052498a","papermill":{"duration":5.522008,"end_time":"2022-01-30T13:02:42.440466","exception":false,"start_time":"2022-01-30T13:02:36.918458","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:19:28.050311Z","iopub.execute_input":"2022-02-13T09:19:28.050678Z","iopub.status.idle":"2022-02-13T09:19:33.666408Z","shell.execute_reply.started":"2022-02-13T09:19:28.050643Z","shell.execute_reply":"2022-02-13T09:19:33.665404Z"},"jupyter":{"source_hidden":true,"outputs_hidden":true},"collapsed":true,"trusted":true},"execution_count":4,"outputs":[]},{"cell_type":"markdown","source":"Next is Stanford's `contrafold` package (ref: http://contra.stanford.edu/contrafold/). Note that it requires `g++4.8` which is not available in the latest Kaggle environment.","metadata":{"papermill":{"duration":0.050075,"end_time":"2022-01-30T13:02:42.542639","exception":false,"start_time":"2022-01-30T13:02:42.492564","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!cp -rf {TOOLKITS_DIR}/contrafold-se/contrafold-se .\n!apt-get install -y g++-4.8\n!sed -i.bak \"1 s/^.*$/CXX = g++-4.8/\" contrafold-se/src/Makefile\n!cd contrafold-se/src; make","metadata":{"_kg_hide-input":true,"_kg_hide-output":true,"papermill":{"duration":31.467915,"end_time":"2022-01-30T13:03:14.061678","exception":false,"start_time":"2022-01-30T13:02:42.593763","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:19:33.671435Z","iopub.execute_input":"2022-02-13T09:19:33.671772Z","iopub.status.idle":"2022-02-13T09:20:05.249888Z","shell.execute_reply.started":"2022-02-13T09:19:33.671731Z","shell.execute_reply":"2022-02-13T09:20:05.249013Z"},"jupyter":{"source_hidden":true,"outputs_hidden":true},"collapsed":true,"trusted":true},"execution_count":5,"outputs":[]},{"cell_type":"markdown","source":"Lastly, the`rnastructure` package from https://rna.urmc.rochester.edu/RNAstructure.html","metadata":{"papermill":{"duration":0.092967,"end_time":"2022-01-30T13:03:14.233653","exception":false,"start_time":"2022-01-30T13:03:14.140686","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!tar zxvf {TOOLKITS_DIR}/RNAstructureLinuxTextInterfaces64bit.tgz --directory .","metadata":{"_kg_hide-input":true,"_kg_hide-output":true,"papermill":{"duration":3.814991,"end_time":"2022-01-30T13:03:18.140255","exception":false,"start_time":"2022-01-30T13:03:14.325264","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:05.253834Z","iopub.execute_input":"2022-02-13T09:20:05.254289Z","iopub.status.idle":"2022-02-13T09:20:09.012320Z","shell.execute_reply.started":"2022-02-13T09:20:05.254251Z","shell.execute_reply":"2022-02-13T09:20:09.011199Z"},"jupyter":{"source_hidden":true,"outputs_hidden":true},"collapsed":true,"trusted":true},"execution_count":6,"outputs":[]},{"cell_type":"markdown","source":"# 2. Feature Extraction\n\nIn order to use `arnie` as an interface to all bpps libraries, we need to configure the paths to `arnie.conf` as shown below. After that, we import all necessary packages that will be used in this tutorial.","metadata":{"papermill":{"duration":0.112814,"end_time":"2022-01-30T13:03:18.376757","exception":false,"start_time":"2022-01-30T13:03:18.263943","status":"completed"},"tags":[]}},{"cell_type":"code","source":"import os\nimport glob\nimport sys\n\n!echo \"vienna_2: /usr/bin\" > arnie.conf\n!echo \"contrafold_2: /kaggle/working/contrafold-se/src\" >> arnie.conf\n!echo \"rnastructure: /kaggle/working/RNAstructure/exe\" >> arnie.conf\n!echo \"TMP: /kaggle/working/tmp\" >> arnie.conf\n!mkdir -p /kaggle/working/tmp\nos.environ[\"ARNIEFILE\"] = f\"/kaggle/working/arnie.conf\"\nos.environ[\"DATAPATH\"] = f\"/kaggle/working/RNAstructure/data_tables\"\nsys.path.append('/kaggle/working/draw_rna_pkg/')\nsys.path.append('/kaggle/working/draw_rna_pkg/ipynb/')\n\n!cat arnie.conf","metadata":{"_kg_hide-output":true,"papermill":{"duration":4.646578,"end_time":"2022-01-30T13:03:23.146696","exception":false,"start_time":"2022-01-30T13:03:18.500118","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:09.014149Z","iopub.execute_input":"2022-02-13T09:20:09.014562Z","iopub.status.idle":"2022-02-13T09:20:13.563584Z","shell.execute_reply.started":"2022-02-13T09:20:09.014515Z","shell.execute_reply":"2022-02-13T09:20:13.562327Z"},"trusted":true},"execution_count":7,"outputs":[]},{"cell_type":"code","source":"import numpy as np\nimport pandas as pd\nfrom tqdm.notebook import tqdm as tqdm\nimport subprocess\nfrom multiprocessing import Pool\n\n# from arnie.pfunc import pfunc\nfrom arnie.mea.mea import MEA\n# from arnie.free_energy import free_energy\nfrom arnie.bpps import bpps\n# from arnie.mfe import mfe\nimport arnie.utils as utils\n\n","metadata":{"papermill":{"duration":0.147558,"end_time":"2022-01-30T13:03:23.40173","exception":false,"start_time":"2022-01-30T13:03:23.254172","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:13.565538Z","iopub.execute_input":"2022-02-13T09:20:13.566103Z","iopub.status.idle":"2022-02-13T09:20:13.597362Z","shell.execute_reply.started":"2022-02-13T09:20:13.566040Z","shell.execute_reply":"2022-02-13T09:20:13.596283Z"},"trusted":true},"execution_count":8,"outputs":[]},{"cell_type":"markdown","source":"## 2.1 Getting Edge Features\n\nEqupping with packages like `vienna`, `contrafold` and `rnastructure`,`arnie` can be used to extract both bpps matrix and most-probable structure.\nHere, we use RNA strings from the OpenVaccine dataset without using any already-extracted-features. \n\nThe readers can easily substitute their own RNA strings to the ones in OpenVaccine by creating a pandas `DataFrame` with two columns of `['id','sequence']`. Note that `id` for each string can be arbitrary -- just make sure to be unique for each string.\n","metadata":{"papermill":{"duration":0.116282,"end_time":"2022-01-30T13:03:23.639174","exception":false,"start_time":"2022-01-30T13:03:23.522892","status":"completed"},"tags":[]}},{"cell_type":"code","source":"debug = True # To extract features of all 6,000 strings, set debug=False","metadata":{"papermill":{"duration":0.115348,"end_time":"2022-01-30T13:03:23.860038","exception":false,"start_time":"2022-01-30T13:03:23.74469","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:13.599100Z","iopub.execute_input":"2022-02-13T09:20:13.599519Z","iopub.status.idle":"2022-02-13T09:20:13.721113Z","shell.execute_reply.started":"2022-02-13T09:20:13.599465Z","shell.execute_reply":"2022-02-13T09:20:13.719462Z"},"trusted":true},"execution_count":9,"outputs":[]},{"cell_type":"code","source":"# extract features from all packages and save in corresponding dir\n# you can also select only one package\npackages = ['contrafold_2', 'vienna_2', 'rnastructure']\n\ntrain = pd.read_json('../input/stanford-covid-vaccine/train.json', lines=True)\ntest = pd.read_json('../input/stanford-covid-vaccine/test.json', lines=True)\n\nif debug:\n    train = train[:10]\n    test = test[:10]\ntarget_df = train.append(test)[['id','sequence']]\n\nprint(target_df.shape)\ntarget_df.head(3)","metadata":{"papermill":{"duration":1.294778,"end_time":"2022-01-30T13:03:25.261042","exception":false,"start_time":"2022-01-30T13:03:23.966264","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:13.722920Z","iopub.execute_input":"2022-02-13T09:20:13.723337Z","iopub.status.idle":"2022-02-13T09:20:14.971879Z","shell.execute_reply.started":"2022-02-13T09:20:13.723303Z","shell.execute_reply":"2022-02-13T09:20:14.970719Z"},"trusted":true},"execution_count":10,"outputs":[]},{"cell_type":"markdown","source":"As Kaggle notebook provides 4-cores CPU, we will take advantage of the multiprocessing when extracting edge/node features. Here, we creat directory for each selected package separately.","metadata":{"papermill":{"duration":0.108179,"end_time":"2022-01-30T13:03:25.480713","exception":false,"start_time":"2022-01-30T13:03:25.372534","status":"completed"},"tags":[]}},{"cell_type":"code","source":"!grep processor /proc/cpuinfo | wc -l\nMAX_THREADS = 4\nBPPS_DIR = WORKING_DIR+'bpps/'\nos.mkdir(BPPS_DIR) # we will store the extracted bpps here","metadata":{"papermill":{"duration":0.875097,"end_time":"2022-01-30T13:03:26.463333","exception":false,"start_time":"2022-01-30T13:03:25.588236","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:14.973600Z","iopub.execute_input":"2022-02-13T09:20:14.974176Z","iopub.status.idle":"2022-02-13T09:20:15.726283Z","shell.execute_reply.started":"2022-02-13T09:20:14.974128Z","shell.execute_reply":"2022-02-13T09:20:15.725175Z"},"trusted":true},"execution_count":11,"outputs":[]},{"cell_type":"code","source":"for package in packages:\n    os.mkdir(BPPS_DIR+package)\n!ls {BPPS_DIR}","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:15.727766Z","iopub.execute_input":"2022-02-13T09:20:15.728200Z","iopub.status.idle":"2022-02-13T09:20:16.490747Z","shell.execute_reply.started":"2022-02-13T09:20:15.728168Z","shell.execute_reply":"2022-02-13T09:20:16.489809Z"},"trusted":true},"execution_count":12,"outputs":[]},{"cell_type":"markdown","source":"The parallel computation can be done easily by applying `imap` from `multiprocessing` package to the specified function, which in our case is the `extract_edge_features` function. `arnie` provides `bpps` and `MEA` function to extract the bpps matrix and most-probable structure, respectively.\nThe extracted bpps matrix will be saved to Numpy's `.npy` format.","metadata":{"papermill":{"duration":0.114825,"end_time":"2022-01-30T13:03:26.692623","exception":false,"start_time":"2022-01-30T13:03:26.577798","status":"completed"},"tags":[]}},{"cell_type":"code","source":"def extract_edge_features(arg):\n    sequence, seq_id, package = arg\n    \n    bp_matrix = bpps(sequence, package=package)\n        \n    mea_mdl = MEA(bp_matrix)\n    np.save(BPPS_DIR+package+f'/{seq_id}.npy', bp_matrix)\n    \n    return seq_id, sequence, mea_mdl.structure, mea_mdl.score_expected()[2], package\n","metadata":{"_kg_hide-output":true,"papermill":{"duration":0.119259,"end_time":"2022-01-30T13:03:26.922674","exception":false,"start_time":"2022-01-30T13:03:26.803415","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:16.492128Z","iopub.execute_input":"2022-02-13T09:20:16.492464Z","iopub.status.idle":"2022-02-13T09:20:16.502285Z","shell.execute_reply.started":"2022-02-13T09:20:16.492429Z","shell.execute_reply":"2022-02-13T09:20:16.501475Z"},"trusted":true},"execution_count":13,"outputs":[]},{"cell_type":"code","source":"# prepare input for extract_edge_features\narg_list = []\nfor i, (seq, seq_id) in enumerate(target_df[['sequence','id']].values):\n    for pack in packages:\n        arg_list.append([seq, seq_id, pack])\n\n# apply multiprocessing to extract_edge_features\np = Pool(processes=MAX_THREADS)\nresults = []\nfor ret in tqdm(p.imap(extract_edge_features, arg_list),total=len(arg_list)):\n    results.append(ret)\n    \nstructure_df = pd.DataFrame(results, columns=['id', 'sequence', 'structure', 'score', 'package'])","metadata":{"papermill":{"duration":5.680838,"end_time":"2022-01-30T13:03:32.710519","exception":false,"start_time":"2022-01-30T13:03:27.029681","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:16.503498Z","iopub.execute_input":"2022-02-13T09:20:16.503922Z","iopub.status.idle":"2022-02-13T09:20:35.884103Z","shell.execute_reply.started":"2022-02-13T09:20:16.503880Z","shell.execute_reply":"2022-02-13T09:20:35.882960Z"},"trusted":true},"execution_count":14,"outputs":[]},{"cell_type":"code","source":"structure_df.head(10)","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:35.886177Z","iopub.execute_input":"2022-02-13T09:20:35.886648Z","iopub.status.idle":"2022-02-13T09:20:35.906474Z","shell.execute_reply.started":"2022-02-13T09:20:35.886611Z","shell.execute_reply":"2022-02-13T09:20:35.905378Z"},"trusted":true},"execution_count":15,"outputs":[]},{"cell_type":"markdown","source":" While the most-probable structures of all strings for each package will be stored separately in a `DataFrame`, namely, `structure_df_list` for easy usage. Each element of `structure_df_list` corresponds to one package as shown below.","metadata":{}},{"cell_type":"code","source":"structure_df_list = []\nfor package in packages:\n    structure_df_list.append(structure_df[structure_df['package']==package])\n    print(structure_df_list[-1].shape)\n","metadata":{"papermill":{"duration":0.135863,"end_time":"2022-01-30T13:03:33.170659","exception":false,"start_time":"2022-01-30T13:03:33.034796","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:35.908015Z","iopub.execute_input":"2022-02-13T09:20:35.908352Z","iopub.status.idle":"2022-02-13T09:20:35.924165Z","shell.execute_reply.started":"2022-02-13T09:20:35.908315Z","shell.execute_reply":"2022-02-13T09:20:35.923338Z"},"trusted":true},"execution_count":16,"outputs":[]},{"cell_type":"markdown","source":"Now, we can investigate the resulted most-probable structure for each string here! The structure is encoded in [a **Dot-Bracket** format](https://www.tbi.univie.ac.at/RNA/ViennaRNA/doc/html/rna_structure_notations.html). The Dot-Bracket notation as introduced in the early times of the ViennaRNA Package denotes base pairs by matching pairs of pair () and unpaired nucleotides by dots. Note that most probable structures of the same id but different packages may not be the same.","metadata":{}},{"cell_type":"code","source":"for i, package in enumerate(packages):\n    print(structure_df_list[i].head(3))","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:35.925473Z","iopub.execute_input":"2022-02-13T09:20:35.926001Z","iopub.status.idle":"2022-02-13T09:20:35.947416Z","shell.execute_reply.started":"2022-02-13T09:20:35.925952Z","shell.execute_reply":"2022-02-13T09:20:35.946152Z"},"trusted":true},"execution_count":17,"outputs":[]},{"cell_type":"markdown","source":"If the above print() command is not so intutive, we can also visualize the most-probable structure easily  🤩, even though we won't use the visualization in our feature extraction procedure.","metadata":{"papermill":{"duration":0.108708,"end_time":"2022-01-30T13:03:33.391222","exception":false,"start_time":"2022-01-30T13:03:33.282514","status":"completed"},"tags":[]}},{"cell_type":"code","source":"from draw import draw_struct\nsequence, structure = structure_df.iloc[0][['sequence', 'structure']]\nprint(sequence)\nprint(structure)\ndraw_struct(sequence, structure,  cmap='plasma')","metadata":{"papermill":{"duration":1.096692,"end_time":"2022-01-30T13:03:34.596366","exception":false,"start_time":"2022-01-30T13:03:33.499674","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:35.948766Z","iopub.execute_input":"2022-02-13T09:20:35.949096Z","iopub.status.idle":"2022-02-13T09:20:36.917125Z","shell.execute_reply.started":"2022-02-13T09:20:35.949063Z","shell.execute_reply":"2022-02-13T09:20:36.916123Z"},"trusted":true},"execution_count":18,"outputs":[]},{"cell_type":"code","source":"# Finally, let us investigate the saved bpps files\nfor i, package in enumerate(packages):\n    print(package)\n    !ls {BPPS_DIR}{package} | wc\n    !ls {BPPS_DIR}{package} | head","metadata":{"papermill":{"duration":1.638721,"end_time":"2022-01-30T13:03:36.347771","exception":false,"start_time":"2022-01-30T13:03:34.70905","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:36.918688Z","iopub.execute_input":"2022-02-13T09:20:36.919017Z","iopub.status.idle":"2022-02-13T09:20:41.439373Z","shell.execute_reply.started":"2022-02-13T09:20:36.918987Z","shell.execute_reply":"2022-02-13T09:20:41.438357Z"},"trusted":true},"execution_count":19,"outputs":[]},{"cell_type":"markdown","source":"## 2.2 Getting Node Features\n\nNow we will extract one-hot ACGU vector as well as the probabilisttic loop-type vector using `CapR` package. The`CapR` is a shell script command which could be run one string at a time with specified input and output files, and we automate the run in the `run_CapR` function and store the probabilistic output with a pandas' dataframe.\n","metadata":{"papermill":{"duration":0.111093,"end_time":"2022-01-30T13:03:36.570257","exception":false,"start_time":"2022-01-30T13:03:36.459164","status":"completed"},"tags":[]}},{"cell_type":"code","source":"NODE_DIR = WORKING_DIR+'node_features/'\nos.mkdir(NODE_DIR)","metadata":{"papermill":{"duration":0.119744,"end_time":"2022-01-30T13:03:36.801204","exception":false,"start_time":"2022-01-30T13:03:36.68146","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.440657Z","iopub.execute_input":"2022-02-13T09:20:41.441002Z","iopub.status.idle":"2022-02-13T09:20:41.446411Z","shell.execute_reply.started":"2022-02-13T09:20:41.440958Z","shell.execute_reply":"2022-02-13T09:20:41.445667Z"},"trusted":true},"execution_count":20,"outputs":[]},{"cell_type":"code","source":"def run_CapR(rna_id, rna_string, max_seq_len=1024):\n    in_file = '%s.fa' % rna_id\n    out_file = '%s.out' % rna_id\n             \n    fp = open(in_file, \"w\")\n    fp.write('>%s\\n' % rna_id)\n    fp.write(rna_string)\n    fp.close()\n    \n    subprocess.run('/kaggle/working/CapR/CapR %s %s %d' % (in_file, out_file, max_seq_len),\n                   shell=True,capture_output=False)\n             \n    df = pd.read_csv(out_file, skiprows=1,\n                     header=None, delim_whitespace=True,\n            )\n    df2 = df.T[1:]\n    df2.columns = df.T.iloc[0].values\n    \n    return df2","metadata":{"_kg_hide-output":true,"papermill":{"duration":0.124886,"end_time":"2022-01-30T13:03:37.03715","exception":false,"start_time":"2022-01-30T13:03:36.912264","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.448091Z","iopub.execute_input":"2022-02-13T09:20:41.448794Z","iopub.status.idle":"2022-02-13T09:20:41.458655Z","shell.execute_reply.started":"2022-02-13T09:20:41.448731Z","shell.execute_reply":"2022-02-13T09:20:41.457819Z"},"trusted":true},"execution_count":21,"outputs":[]},{"cell_type":"markdown","source":"We can test run the function just to have fun :)","metadata":{"papermill":{"duration":0.123884,"end_time":"2022-01-30T13:03:37.273768","exception":false,"start_time":"2022-01-30T13:03:37.149884","status":"completed"},"tags":[]}},{"cell_type":"code","source":"test_string = 'AGGGUUUUCCCC'\ndf = run_CapR('test_id_1234',test_string )\ndf.head(len(test_string ))","metadata":{"papermill":{"duration":0.164438,"end_time":"2022-01-30T13:03:37.556475","exception":false,"start_time":"2022-01-30T13:03:37.392037","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.460071Z","iopub.execute_input":"2022-02-13T09:20:41.460620Z","iopub.status.idle":"2022-02-13T09:20:41.669567Z","shell.execute_reply.started":"2022-02-13T09:20:41.460574Z","shell.execute_reply":"2022-02-13T09:20:41.668686Z"},"trusted":true},"execution_count":22,"outputs":[]},{"cell_type":"markdown","source":"Next, `extract_rna_node_features` will combine the above `run_CapR` with one-hot 'AGUC' vector. We also can have the most-probable structure in DotBracket notation as one-hot vector as an additional option to node features. This `extract_rna_node_features` will save a `.csv` file for each string in the `NODE_DIR` directory","metadata":{"papermill":{"duration":0.110571,"end_time":"2022-01-30T13:03:37.793415","exception":false,"start_time":"2022-01-30T13:03:37.682844","status":"completed"},"tags":[]}},{"cell_type":"code","source":"def extract_rna_node_features(rna_id, rna_string, mfe_structure=None):\n    # looptype features\n    df = run_CapR(rna_id, rna_string)\n    \n    # onehot AGCU features\n    def onehot_np(length, i):\n        vect = [0] * length\n        vect[i] = 1\n        return np.array(vect)\n    \n    base_vocab = 'ACGU'\n    token2onehot = {x:onehot_np(len(base_vocab), i) for i, x in enumerate(base_vocab)}\n    out = list(map(lambda y : token2onehot[y], rna_string))\n    df[list(base_vocab)] = np.array(out)\n    \n    # onehot mfe_structure features (optional)\n    if mfe_structure is not None:\n        structure_vocab = '(.)'\n        token2onehot = {x:onehot_np(len(structure_vocab), i) for i, x in enumerate(structure_vocab)}\n        out = list(map(lambda y : token2onehot[y], mfe_structure))\n        df[list(structure_vocab)] = np.array(out)\n    \n    return df","metadata":{"papermill":{"duration":0.283902,"end_time":"2022-01-30T13:03:38.189966","exception":false,"start_time":"2022-01-30T13:03:37.906064","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.670922Z","iopub.execute_input":"2022-02-13T09:20:41.671413Z","iopub.status.idle":"2022-02-13T09:20:41.682257Z","shell.execute_reply.started":"2022-02-13T09:20:41.671380Z","shell.execute_reply":"2022-02-13T09:20:41.681378Z"},"trusted":true},"execution_count":23,"outputs":[]},{"cell_type":"markdown","source":"Run for another fun to see the complete sequence of node features for each character 🚀 💥 🚀!! ","metadata":{"papermill":{"duration":0.111266,"end_time":"2022-01-30T13:03:38.414018","exception":false,"start_time":"2022-01-30T13:03:38.302752","status":"completed"},"tags":[]}},{"cell_type":"code","source":"rna_id = 'testmol'\nrna_string = 'AGGGGCCUUUUAAGGAAUUUC'\nstruct_string = '(.................)()'\n\ndf = extract_rna_node_features(rna_id, rna_string, struct_string)\nprint(df.shape)\ndf.head(10)","metadata":{"papermill":{"duration":0.168469,"end_time":"2022-01-30T13:03:38.695666","exception":false,"start_time":"2022-01-30T13:03:38.527197","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.683682Z","iopub.execute_input":"2022-02-13T09:20:41.684057Z","iopub.status.idle":"2022-02-13T09:20:41.742214Z","shell.execute_reply.started":"2022-02-13T09:20:41.684026Z","shell.execute_reply":"2022-02-13T09:20:41.741085Z"},"trusted":true},"execution_count":24,"outputs":[]},{"cell_type":"markdown","source":"Now we make a simple function to allow parallelization and save the resulted node features into an individual dataframe. Note that here, use only 1 most-probable structure from the first package to make a feature, but the reader can extend to include structures from all packages easily","metadata":{"papermill":{"duration":0.111915,"end_time":"2022-01-30T13:03:38.921435","exception":false,"start_time":"2022-01-30T13:03:38.80952","status":"completed"},"tags":[]}},{"cell_type":"code","source":"def extract_rna_node_features_and_save(arg):\n    rna_id = arg[0]\n    rna_string = arg[1]\n    struct_string = arg[2]\n\n    df = extract_rna_node_features(rna_id, rna_string, struct_string)\n    df.to_csv(NODE_DIR+'%s_node_features.csv' % rna_id, index=False)\n    \n    return 0","metadata":{"papermill":{"duration":0.123808,"end_time":"2022-01-30T13:03:39.20892","exception":false,"start_time":"2022-01-30T13:03:39.085112","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.744324Z","iopub.execute_input":"2022-02-13T09:20:41.744773Z","iopub.status.idle":"2022-02-13T09:20:41.751281Z","shell.execute_reply.started":"2022-02-13T09:20:41.744722Z","shell.execute_reply":"2022-02-13T09:20:41.750258Z"},"trusted":true},"execution_count":25,"outputs":[]},{"cell_type":"code","source":"arg_list = []\n\nfor i, (seq_id, sequence, structure) in enumerate(structure_df_list[0][['id','sequence','structure']].values):\n#     for pack in packages:\n    arg_list.append([seq_id, sequence, structure])\n            \np = Pool(processes=MAX_THREADS)\n\nfor ret in tqdm(p.imap(extract_rna_node_features_and_save, arg_list),total=len(arg_list)):\n    pass # we save the output to files","metadata":{"papermill":{"duration":1.403374,"end_time":"2022-01-30T13:03:40.725847","exception":false,"start_time":"2022-01-30T13:03:39.322473","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:41.752626Z","iopub.execute_input":"2022-02-13T09:20:41.753150Z","iopub.status.idle":"2022-02-13T09:20:43.030537Z","shell.execute_reply.started":"2022-02-13T09:20:41.753116Z","shell.execute_reply":"2022-02-13T09:20:43.029527Z"},"trusted":true},"execution_count":26,"outputs":[]},{"cell_type":"markdown","source":"Node-feature extraction is done ☄️ 💥 🔥 !!","metadata":{"papermill":{"duration":0.116478,"end_time":"2022-01-30T13:03:40.95895","exception":false,"start_time":"2022-01-30T13:03:40.842472","status":"completed"},"tags":[]}},{"cell_type":"code","source":"# investigate the save bpps files\n!ls {NODE_DIR} | wc\n!ls {NODE_DIR} | head","metadata":{"papermill":{"duration":1.645226,"end_time":"2022-01-30T13:03:42.719847","exception":false,"start_time":"2022-01-30T13:03:41.074621","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:43.032286Z","iopub.execute_input":"2022-02-13T09:20:43.032705Z","iopub.status.idle":"2022-02-13T09:20:44.576591Z","shell.execute_reply.started":"2022-02-13T09:20:43.032673Z","shell.execute_reply":"2022-02-13T09:20:44.575511Z"},"trusted":true},"execution_count":27,"outputs":[]},{"cell_type":"markdown","source":"# (Optional) 3. Advanced Node-Feature Extraction\n\nWe are done, but what else? We can actually go beyond to explore some feature engineering like SOTA [models](https://www.kaggle.com/group16/covid-19-mrna-4th-place-solution), which also was inspried by [this](https://www.kaggle.com/its7171/gru-lstm-with-feature-engineering-and-augmentation) and this [notebooks](https://www.kaggle.com/its7171/dangerous-features). Yes, SOTA models were those who got gold medals in the very-intense OpenVaccine competition!\n\nAdvanced node features that we are going to extract further in this section are\n\n* The ratio of Dot-Bracket prediction `'(', '.',` and `')'` predicted by 3 packages (`vienna`, `contrafold` and `rnastructure`) -- instead of depending only on one package like we did in previous section\n* Some of bpps for each base\n* Maximum probability with respect to bpps for each base\n* Number of non-zeros on bpps for each base\n* Codon's (triplet) position for each base which is simply 012012012012... for every RNA string.\n","metadata":{}},{"cell_type":"code","source":"# we will save these advanced node features separated from basic node features, so that users can choose either of them\nADV_NODE_DIR = '/kaggle/working/advanced_node_features/'\nos.mkdir(ADV_NODE_DIR) ","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:44.578754Z","iopub.execute_input":"2022-02-13T09:20:44.579254Z","iopub.status.idle":"2022-02-13T09:20:44.585439Z","shell.execute_reply.started":"2022-02-13T09:20:44.579206Z","shell.execute_reply":"2022-02-13T09:20:44.584429Z"},"trusted":true},"execution_count":28,"outputs":[]},{"cell_type":"code","source":"rna_id_list = structure_df.id.unique()\nstructure_df.head(2)","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:44.587285Z","iopub.execute_input":"2022-02-13T09:20:44.588033Z","iopub.status.idle":"2022-02-13T09:20:44.610502Z","shell.execute_reply.started":"2022-02-13T09:20:44.587979Z","shell.execute_reply":"2022-02-13T09:20:44.609391Z"},"trusted":true},"execution_count":29,"outputs":[]},{"cell_type":"code","source":"def str_to_charflag(string, char):\n    bin_str = [c==char for c in string]\n    return np.array(bin_str).astype(np.int32)\n\n# this function assumes the existent of structure_df built in last section\ndef make_advanced_features(rna_id,debug=False):\n    # first, let us copy the basic features    \n    df = pd.read_csv(NODE_DIR+rna_id+'_node_features.csv')\n    \n    ####\n    # Ratio of the open-pair, close-pair and non-pair\n    ####\n    struct_string_list = structure_df.query('id == @rna_id')['structure'].values\n    \n    open_pair_list = np.array([str_to_charflag(s, '(') for s in struct_string_list])\n    open_pair_ratio = np.mean(open_pair_list, axis=0)\n    df['(-ratio'] = open_pair_ratio\n    \n    close_pair_list = np.array([str_to_charflag(s, ')') for s in struct_string_list])\n    close_pair_ratio = np.mean(close_pair_list, axis=0)\n    df[')-ratio'] = close_pair_ratio\n    \n    non_pair_list = np.array([str_to_charflag(s, '.') for s in struct_string_list])\n    non_pair_ratio = np.mean(non_pair_list, axis=0)\n    df['.-ratio'] = non_pair_ratio\n    \n    ####\n    # Codon positioning feature (trinary)\n    ####\n    rna_len = len(struct_string_list[0])\n    codon_pos0 = (np.arange(rna_len) % 3 == 0).astype(np.int32)\n    codon_pos1 = (np.arange(rna_len) % 3 == 1).astype(np.int32)\n    codon_pos2 = (np.arange(rna_len) % 3 == 2).astype(np.int32)\n    df['codon-pos0'] = codon_pos0\n    df['codon-pos1'] = codon_pos1\n    df['codon-pos2'] = codon_pos2\n    \n    ####\n    # bpps feat. engineering\n    ####\n    bpps_list = []\n    for i, package in enumerate(packages):\n        bpps_list.append(np.load(BPPS_DIR+package+f'/{rna_id}.npy'))\n    bpps = np.mean(bpps_list, axis=0)\n    \n    df['bpps-max'] = np.max(bpps,axis=0)\n    df['bpps-sum'] = np.sum(bpps,axis=0)\n    df['bpps-num-ratio'] = np.sum(bpps==0, axis=0)/bpps.shape[0]\n    \n    df.to_csv(ADV_NODE_DIR+'%s_node_features.csv' % rna_id, index=False)\n    \n    if debug:\n        return df\n    return 0\n\ndf = make_advanced_features(rna_id_list[0], debug=True)\nprint(df.shape)\ndf.head(25)","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:44.612320Z","iopub.execute_input":"2022-02-13T09:20:44.612989Z","iopub.status.idle":"2022-02-13T09:20:44.738194Z","shell.execute_reply.started":"2022-02-13T09:20:44.612924Z","shell.execute_reply":"2022-02-13T09:20:44.736847Z"},"trusted":true},"execution_count":30,"outputs":[]},{"cell_type":"code","source":"p = Pool(processes=MAX_THREADS)\n\nfor ret in tqdm(p.imap(make_advanced_features, rna_id_list),total=len(rna_id_list)):\n    pass # we save the output to files","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:44.739920Z","iopub.execute_input":"2022-02-13T09:20:44.740359Z","iopub.status.idle":"2022-02-13T09:20:45.047851Z","shell.execute_reply.started":"2022-02-13T09:20:44.740315Z","shell.execute_reply":"2022-02-13T09:20:45.046398Z"},"trusted":true},"execution_count":31,"outputs":[]},{"cell_type":"code","source":"# investigate the save bpps files\n!ls {ADV_NODE_DIR} | wc\n!ls {ADV_NODE_DIR} | head","metadata":{"execution":{"iopub.status.busy":"2022-02-13T09:20:45.056400Z","iopub.execute_input":"2022-02-13T09:20:45.057248Z","iopub.status.idle":"2022-02-13T09:20:46.642586Z","shell.execute_reply.started":"2022-02-13T09:20:45.057191Z","shell.execute_reply":"2022-02-13T09:20:46.641447Z"},"trusted":true},"execution_count":32,"outputs":[]},{"cell_type":"markdown","source":"# Epilogue: Clean up before finish!!\nTo be compact, we will zip all the extract features and remove all the packages before leaving.\nBy this clean up, it will easy to export into another Kaggle's dataset to be used in another Deep-Modeling notebook\n\nThis concludes the tutorial; hope it will be useful 🥰 😘!!","metadata":{"papermill":{"duration":0.114964,"end_time":"2022-01-30T13:03:42.948766","exception":false,"start_time":"2022-01-30T13:03:42.833802","status":"completed"},"tags":[]}},{"cell_type":"code","source":"structure_df.to_csv('most_probable_structure.csv',index=False)\n!zip -r bpps.zip {BPPS_DIR}\n!zip -r node_features.zip {NODE_DIR}\n!zip -r advanced_node_features.zip {ADV_NODE_DIR}","metadata":{"_kg_hide-output":true,"papermill":{"duration":1.692194,"end_time":"2022-01-30T13:03:44.756637","exception":false,"start_time":"2022-01-30T13:03:43.064443","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:46.644816Z","iopub.execute_input":"2022-02-13T09:20:46.645299Z","iopub.status.idle":"2022-02-13T09:20:49.079603Z","shell.execute_reply.started":"2022-02-13T09:20:46.645262Z","shell.execute_reply":"2022-02-13T09:20:49.078518Z"},"collapsed":true,"jupyter":{"outputs_hidden":true},"trusted":true},"execution_count":33,"outputs":[]},{"cell_type":"code","source":"!rm -rf CapR/\n!rm -r tmp/\n!rm -r RNAstructure/\n!rm -r arnie/\n!rm -r contrafold-se/\n!rm -r draw_rna_pkg/\n!rm -rf {BPPS_DIR}\n!rm -rf {NODE_DIR}\n!rm -rf {ADV_NODE_DIR}\n!rm arnie.conf","metadata":{"papermill":{"duration":7.199379,"end_time":"2022-01-30T13:03:52.071103","exception":false,"start_time":"2022-01-30T13:03:44.871724","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:49.081628Z","iopub.execute_input":"2022-02-13T09:20:49.082293Z","iopub.status.idle":"2022-02-13T09:20:56.888904Z","shell.execute_reply.started":"2022-02-13T09:20:49.082241Z","shell.execute_reply":"2022-02-13T09:20:56.887511Z"},"trusted":true},"execution_count":34,"outputs":[]},{"cell_type":"code","source":"# delete temp files relating CapR\nfor f in glob.glob(\"*.fa\"):\n    os.remove(f)\nfor f in glob.glob(\"*.out\"):\n    os.remove(f)\n\n!ls -h","metadata":{"papermill":{"duration":0.884755,"end_time":"2022-01-30T13:03:53.073652","exception":false,"start_time":"2022-01-30T13:03:52.188897","status":"completed"},"tags":[],"execution":{"iopub.status.busy":"2022-02-13T09:20:56.891018Z","iopub.execute_input":"2022-02-13T09:20:56.891443Z","iopub.status.idle":"2022-02-13T09:20:57.660512Z","shell.execute_reply.started":"2022-02-13T09:20:56.891403Z","shell.execute_reply":"2022-02-13T09:20:57.659390Z"},"trusted":true},"execution_count":35,"outputs":[]},{"cell_type":"code","source":"","metadata":{"papermill":{"duration":0.120337,"end_time":"2022-01-30T13:03:53.311982","exception":false,"start_time":"2022-01-30T13:03:53.191645","status":"completed"},"tags":[]},"execution_count":null,"outputs":[]}]}